src/main/java/ch/njol/skript/expressions/ExprBossBarTitle.java

```diff
@@ -0,0 +1,73 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Boss Bar Title")
+@Description("The title/text of a boss bar. This is the text shown above the progress bar of the bossbar.")
+@Examples({"set title of player's bossbar to \"Goodbye!\""})
+@Since("INSERT VERSION")
+public class ExprBossBarTitle extends SimplePropertyExpression<BossBar, String> {
+
+	static {
+		register(ExprBossBarTitle.class, String.class, "title", "bossbars");
+	}
+
+	@Override
+	@Nullable
+	public String convert(final BossBar bossBar) {
+		return bossBar.getTitle();
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "title";
+	}
+
+	@Override
+	public Class<String> getReturnType() {
+		return String.class;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(final Changer.ChangeMode mode) {
+		if (mode == Changer.ChangeMode.SET)
+			return new Class[] {String.class};
+		return null;
+	}
+
+	@Override
+	public void change(final Event event, final @Nullable Object[] delta, final Changer.ChangeMode mode) {
+		for (final BossBar bossBar : getExpr().getArray(event)) {
+			assert delta[0] != null;
```

Comment:
It would be better to have the assert check and the cast to String before the for loop